pkg/gpu/mocknvml/engine/handles.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package engine
+
+import (
+	"sync"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+// HandleTable manages the mapping between C handles (uintptr) and Go device
+// objects. This is necessary because CGo doesn't allow passing Go pointers
+// with nested Go pointers to C code.
+type HandleTable struct {
+	devices map[uintptr]nvml.Device
+	reverse map[nvml.Device]uintptr
+	counter uintptr
+	mu      sync.RWMutex
+}
+
+// NewHandleTable creates a new HandleTable.
+func NewHandleTable() *HandleTable {
+	return &HandleTable{
+		devices: make(map[uintptr]nvml.Device),
+		reverse: make(map[nvml.Device]uintptr),
+		counter: 1, // Start from 1, 0 reserved for NULL
```

Comment:
When is this counter actually used? Why is it not sufficient to check `len(devices)` and `len(reverse)`?